src/Integration.UnitTests/Helpers/SonarLintOutputTests.cs

```diff
@@ -64,35 +63,95 @@ public void LogVerbose_OutputsToWindowIfLogLevelIsVerbose(DaemonLogLevel logLeve
         {
             // Arrange
             var windowMock = new ConfigurableVsOutputWindow();
+            var serviceProviderMock = CreateConfiguredServiceProvider(windowMock);
 
-            var serviceProviderMock = new ConfigurableServiceProvider(assertOnUnexpectedServiceRequest: true);
-            serviceProviderMock.RegisterService(typeof(SVsOutputWindow), windowMock);
-
-            var sonarLintSettings = new Mock<ISonarLintSettings>();
-            sonarLintSettings.Setup(x => x.DaemonLogLevel).Returns(logLevel);
-
-            var logger = new SonarLintOutputLogger(serviceProviderMock, sonarLintSettings.Object);
+            var sonarLintSettings = CreateSonarLintSettings(logLevel);
 
-            logger.WriteLine("create window pane");
+            var testSubject = CreateTestSubject(serviceProviderMock, sonarLintSettings);
 
+            testSubject.WriteLine("create window pane");
             var outputPane = windowMock.AssertPaneExists(VsShellUtils.SonarLintOutputPaneGuid);
             outputPane.Reset();
 
             // Act
-            logger.LogVerbose("123 {0} {1}", "param 1", 2);
-            logger.LogVerbose("{0} {1} abc", 1, "param 2");
+            testSubject.LogVerbose("123 {0} {1}", "param 1", 2);
+            testSubject.LogVerbose("{0} {1} abc", 1, "param 2");
 
             // Assert
 
             if (shouldLogMessage)
             {
-                outputPane.AssertOutputStrings("DEBUG: 123 param 1 2", "DEBUG: 1 param 2 abc");
+                outputPane.AssertOutputStrings("[DEBUG] 123 param 1 2", "[DEBUG] 1 param 2 abc");
                 outputPane.AssertOutputStrings(2);
             }
             else
             {
                 outputPane.AssertOutputStrings(0);
             }
         }
+
+        [TestMethod]
+        [DataRow(DaemonLogLevel.Info)]
+        [DataRow(DaemonLogLevel.Minimal)]
+        [DataRow(DaemonLogLevel.Verbose)]
+        public void LogVerbose_PrefixIsAddedIfLogLevelIsVerbose(DaemonLogLevel logLevel)
+        {
+            // Arrange
+            var windowMock = new ConfigurableVsOutputWindow();
+            var serviceProviderMock = CreateConfiguredServiceProvider(windowMock);
+
+            var sonarLintSettings = CreateSonarLintSettings(logLevel);
+
+            var testSubject = CreateTestSubject(serviceProviderMock, sonarLintSettings);
+
+            testSubject.WriteLine("create window pane");
+            var outputPane = windowMock.AssertPaneExists(VsShellUtils.SonarLintOutputPaneGuid);
+            outputPane.Reset();
```

Comment:
These tests use custom mock test implementations rather than Moq mocks.
Lines 107-109 are boilerplate to set up the custom classes.

I didn't try to replace the custom classes as part of this change.

Reply:
What is the difference between this and LogVerbose_OutputsToWindowIfLogLevelIsVerbose ?

Reply:
This test checks that the thread id is written out in verbose mode, whichever logging method you use (either `WriteLine` overload, or `LogVerbose`).

The existing test checks that `LogVerbose` only logs in `Verbose` mode.

Reply:
@rita-sonarsource I've renamed and simplified the test so it only checks `WriteLine`.